janusgraph-core/src/main/java/org/janusgraph/graphdb/query/condition/DirectionCondition.java

```diff
@@ -32,9 +33,8 @@
     private final Direction direction;
 
     public DirectionCondition(JanusGraphVertex vertex, Direction dir) {
-        assert vertex != null && dir != null;
```

Comment:
This changes the flow a bit. previously it checked that both vertex and dir were not null. Now it checks vertex, assigned it to `this.baseVertex`, and then checks dir and assigns it to `this.direction`. In the event that vertex is populated, but dir is null we'll get different behavior. Either we need 2 separate null checks at the top, or 1 null check and one in line null check like:

```
Preconditions.checkNotNull(dir);
this.baseVertex = Preconditions.checkNotNull(vertex);
this.direction = dir;
```

Reply:
If a single check fails preconditions throws an exception, so the constructor will fail. The assigned is dir null case only performance change, but exception handling take in the most languages much longer than writing a variable. 

One case i would be if baseVertex is static but this is not the case.